echopype/echodata/echodata.py

```diff
@@ -268,43 +268,42 @@ def __setattr__(self, __name: str, __value: Any) -> None:
     def update_platform(
         self,
         extra_platform_data: xr.Dataset,
-        time_dim="time",
+        variable_mappings=Dict[str, str],
         extra_platform_data_file_name=None,
     ):
         """
         Updates the `EchoData["Platform"]` group with additional external platform data.
 
-        `extra_platform_data` must be an xarray Dataset.
-        The name of the time dimension in `extra_platform_data` is specified by the
-        `time_dim` parameter.
-        Data is extracted from `extra_platform_data` by variable name; only the data
-        in `extra_platform_data` with the following variable names will be used:
-            - `"pitch"`
-            - `"roll"`
-            - `"vertical_offset"`
-            - `"latitude"`
-            - `"longitude"`
-            - `"water_level"`
-        The data inserted into the Platform group will be indexed by a dimension named
-        `"time1"`.
+        `extra_platform_data` must be an xarray Dataset. Data is extracted from
+        `extra_platform_data` by variable name. Only data assigned to a pre-existing
+        (but possibly all-nan) variable name in Platform will be processed. These
+        Platform variables include latitude, longitude, pitch, roll, vertical_offset, etc.
```

Comment:
How about putting a bullet point list here for allowable variables (those in `VALID_PLATFORM_VARS` -- see [my other comment](https://github.com/OSOceanAcoustics/echopype/pull/1060/files#r1232423132)) so people know what these are?

Reply:
Hmm, I'd make the same comment as the one on your other comment. The user could simply inspect the set of variables in their echodata object's Platform group. After all, to apply `update_platform`, the user needs to already have a Platform object. How about adding a statement to that effect in this docstring? Here's a suggestion:
```suggestion
        Platform variables include latitude, longitude, pitch, roll, vertical_offset, etc.
        See the variables present in the EchoData object's Platform group to obtain a
        complete list of possible variables.
```